crates/pop-cli/src/commands/call/contract.rs

```diff
@@ -303,18 +313,25 @@ impl CallContractCommand {
 			self.proof_size = proof_size_input.parse::<u64>().ok(); // If blank or bad input, estimate it.
 		}
 
-		// Resolve who is calling the contract.
-		if self.suri == DEFAULT_URI {
-			// Prompt for uri.
-			self.suri = cli
-				.input("Signer calling the contract:")
-				.placeholder("//Alice")
-				.default_input("//Alice")
-				.interact()?;
-		};
+		// Resolve who is calling the contract. If a `suri` was provided via the command line, skip
+		// the prompt.
+		if self.suri == DEFAULT_URI && !self.use_wallet && message.mutates {
+			if cli.confirm("Do you want to use your browser wallet to sign the transaction? (Selecting 'No' will prompt you to manually enter the secret key URI for signing, e.g., '//Alice')")
+			.initial_value(true)
+			.interact()? {
+				self.use_wallet = true;
+			}
+			else{
+					self.suri = cli
+						.input("Signer calling the contract:")
+						.placeholder("//Alice")
+						.default_input("//Alice")
```

Comment:
Just checking we want the default input to be Alice? Fine for the placeholder.

Reply:
cc @AlexD10S, seems like this is currently required for the flow of call contract?

Reply:
Yes, I think it’s useful to have a default value here. In my experience, most of the time I’ve used `//Alice`.